levelkdev/swapr-sdk
# feat: Curve Factory Pools

- So in this pr we added curve factory pools to the router those are basically pools created by people and not official curve ones
- So changes we made are inside the getRoutablePools function where we added one function which along with curve pools fetches the factory pools thats the easy part
- Then inside the function fetchCurveFactoryPools that fetches the pools we had to map them to fit CurvePool type and dynamically find the abi for the corresponding pool and token type(for the mainnet)
- We also introduced bunch of changes for the abis and found that some pools weren't being routed through and fixes that as well 
- Also for the tri curve pools we made them more dry so as not to repeat the same 3 tokens over and over
- We are also filtering factory pools with liquidity below 100k this is up to debate @adamazad thoughts?

How to test:
1) First checkout to this branch on sdk repo do yarn and then yarn build

2) Open swapr dapp repo and checkout to branch "curve-pools-fix" or to develop if its merged by the time thsi is tested

3) Then you need to connect sdk repo to the local swapr dapp by doing yarn link inside the sdk repo and then inside the swapr dapp do yarn link @swapr/sdk

4) fetch pools from the curve api we use this for mainnet pools https://api.curve.fi/api/getPools/ethereum/factory for the gnosis instead of ethereum type xdai, for arbitrum replace with arbitrum and for polygon replace with polygon

5) Then find the tokens which have above 100k liquidity you can see that by checking the usdTotal property in the api call
<img width="568" alt="Screenshot 2022-07-29 at 19 49 01" src="https://user-images.githubusercontent.com/33226956/181816263-58cc4d49-4f2a-4f14-a786-25ec74897f3a.png">

6) Some tokens can't be found right away so you need to input the address manually into token search to find them
![Screenshot 2022-07-29 at 19 50 47](https://user-images.githubusercontent.com/33226956/181816560-8bb15d68-fac8-4de2-89ca-9114b6041e54.png)

7) Make sure that all pools that have sufficient liquidity can be routed through and keep in mind if its 3crv contains usdt,usdc and dai tokens or 2crv pool which contains usdt and usdc tokens

Feel free to ping me if you have issues with setting all of this up!
